api/src/main/java/com/redhat/ipaas/api/v1/rest/IntegrationDAO.java

```diff
@@ -0,0 +1,151 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.api.model.ConfigMapList;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+@Named(Integration.KIND)
+@ApplicationScoped
+public class IntegrationDAO implements DataAccessObject<Integration> {
+
+
+    private KubernetesClient kubernetesClient;
+
+    public IntegrationDAO() {
+        this(new DefaultKubernetesClient());
+    }
+
+    @Inject
+    public IntegrationDAO(KubernetesClient kubernetesClient) {
+        this.kubernetesClient = kubernetesClient;
+    }
+
+    @Override
+    public Class<Integration> getType() {
+        return Integration.class;
+    }
+
+    @Override
+    public Integration fetch(String id) {
+        return toIntegration(kubernetesClient.configMaps().withName(id).get());
+    }
+
+    @Override
+    public ListResult<Integration> fetchAll() {
+        ConfigMapList list = kubernetesClient.configMaps().withLabel(Integration.LABEL).list();
+        List<ConfigMap> maps = list != null ? list.getItems() : Collections.emptyList();
+        List<Integration> integrations = maps.stream().map(c -> toIntegration(c)).collect(Collectors.toList());
+        return new ListResult.Builder<Integration>().items(integrations).build();
+    }
+
+    @Override
+    public Integration create(Integration entity) {
+        return toIntegration(kubernetesClient.configMaps().create(toConfigMap(entity)));
+    }
+
+    @Override
+    public Integration update(Integration entity) {
+        String id = entity.getId().orElseThrow(IllegalArgumentException::new);
+        ConfigMap old = kubernetesClient.configMaps().withName(id).get();
+
+        kubernetesClient.configMaps().withName(id).replace(toConfigMap(entity));
+        if (old != null) {
+            return toIntegration(old);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public boolean delete(WithId<Integration> entity) {
+        try {
+            return kubernetesClient.configMaps().withName(entity.getId().orElseThrow(IllegalArgumentException::new)).delete();
+        } catch (Throwable t) {
+            throw IPaasServerException.launderThrowable(t);
+        }
+    }
+
+    private static final boolean containsIntegration(ConfigMap configMap) {
+        return configMap != null && configMap.getMetadata().getLabels().containsKey(Integration.LABEL);
+    }
+
+    private static final Integration toIntegration(ConfigMap configMap) {
+        if (configMap == null) {
+            throw new IllegalArgumentException("ConfigMap cannot be null.");
+        } else if (configMap.getMetadata() == null) {
```

Comment:
With your suggestion you have to read inside the block, to understand the flow.

No real difference though, I'll adopt your suggestion.